File: animation.py
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import cm
from mpl_toolkits.mplot3d import Axes3D
import matplotlib.animation as animation

X = Y = np.arange(0, 2, 0.05)
X, Y = np.meshgrid(X, Y)

m = int(input('m: '))
n = int(input('n: '))

frames = 101
Z = [3*(np.sin(m*np.pi*X/2))*(np.sin(n*np.pi*Y/2))*np.sin(i/10) for i in range(frames)]

fig = plt.figure()
ax = Axes3D(fig)
ax.set_zlim(-5, 5)
plot = ax.plot_surface(X, Y, Z[50], cmap=cm.viridis)

def update_lines(frame):
    global plot
    plot.remove()
    plot = ax.plot_surface(X, Y, Z[frame], cmap=cm.viridis)
    return plot,

ani = animation.FuncAnimation(fig, update_lines, frames, interval=50)

plt.show()

File: animation.m
clc
clear all
[x,y] = meshgrid(0:0.05:2,0:.05:2);
m=input('m');
n=input('n')
for i=0:100
z= 3*(sin(m*pi*x/2)).*(sin(n*pi*y/2)).*sin(i/10);
surf(x,y,z);
zlim([-5 5])
pause(0.1)
end

0.100783347396329+6.176529942819232e-04+6.176529942819232e-04+0.100783347396329